xapian-core/weight/bm25weight.cc

```diff
@@ -170,7 +194,15 @@ BM25Weight::get_sumpart(Xapian::termcount wdf, Xapian::termcount len,
     double wdf_double = wdf;
     double denom = param_k1 * (normlen * param_b + (1 - param_b)) + wdf_double;
     AssertRel(denom,>,0);
-    RETURN(termweight * (wdf_double / denom));
+    if(!bm25+) {
+        RETURN(termweight * (wdf_double / denom));
+    }
+    else if(bm25+) {
```

Comment:
You should just have `if (bm25_plus) { … } else { … }`. Testing the negation, then using an `if else` is less readable for people trying to understand what's going on.